lib/lrama/grammar/parameterizing_rule/resolver.rb

```diff
@@ -13,8 +13,14 @@ def add_parameterizing_rule(rule)
           @rules << rule
         end
 
-        def find(token)
-          select_rules(token).last
+        def find_rule(token)
+          if token.is_a?(Lexer::Token::InstantiateRule)
```

Comment:
Is there any reason for this check?
As far as I read the code, caller of this method (https://github.com/ruby/lrama/blob/b08572c2cf4ce0672ffc99d305583362f05f6154/lib/lrama/grammar/rule_builder.rb#L120) ensures `token` is a `InstantiateRule`. If there is not reason to return `nil` for other than `InstantiateRule`, it's better to raise an error whose message includes `BUG`.

Reply:
Indeed it is! Thank you. I updated this PR: https://github.com/ruby/lrama/pull/369/commits/bba8995d6222091cda48b69663e4a6defac6e87b